roc.go

```diff
@@ -0,0 +1,53 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stat
+
+import "sort"
+
+type ByX struct {
+	x []float64
+	c []bool
+}
+
+func (a ByX) Len() int { return len(a.x) }
+func (a ByX) Swap(i, j int) {
+	a.x[i], a.x[j] = a.x[j], a.x[i]
+	a.c[i], a.c[j] = a.c[j], a.c[i]
+}
+func (a ByX) Less(i, j int) bool { return a.x[i] < a.x[j] }
+
+
+// ROC returns paired FPR and TPR values which describe the 
+// ROC curve treating y as a binary classifier for pos.
```

Comment:
also perhaps explain a bit what ROC (receiver operating characteristic), TPR (true positive rate) and FPR (false positive rate) stand for.
you couls also point to wikipedia for some more detailed informations.